src/content/app/entity-viewer/state/general/entityViewerGeneralSlice.ts

```diff
@@ -52,61 +42,46 @@ export const initialState: EntityViewerGeneralState = {
   activeEntityIds: {}
 };
 
+// TODO: rename to setActiveIds? Or setInitialData?
 export const setDataFromUrl =
   (params: EntityViewerParams): ThunkAction<void, any, void, Action<string>> =>
   (dispatch, getState: () => RootState) => {
     const state = getState();
-    const { genomeId: genomeIdFromUrl } = params;
+    const { genomeId, entityId } = params;
 
     let activeGenomeId = getEntityViewerActiveGenomeId(state);
     const activeEntityId = getEntityViewerActiveEntityId(state) || undefined;
 
-    if (!genomeIdFromUrl && !activeGenomeId) {
+    if (!genomeId && !activeGenomeId) {
       dispatch(setDefaultActiveGenomeId());
       activeGenomeId = getEntityViewerActiveGenomeId(state) as string;
-    } else if (genomeIdFromUrl && genomeIdFromUrl !== activeGenomeId) {
-      dispatch(setActiveGenomeId(genomeIdFromUrl));
-      dispatch(fetchGenomeData(genomeIdFromUrl));
-      dispatch(ensureSpeciesIsEnabled(genomeIdFromUrl));
-
-      // TODO: when backend is ready, entity info may also need fetching
-    } else if (activeGenomeId) {
-      // TODO: when backend is ready, fetch entity info
-      const genomeInfo = getGenomeInfoById(state, activeGenomeId);
-      if (!genomeInfo) {
-        dispatch(fetchGenomeData(activeGenomeId));
-      }
-      dispatch(ensureSpeciesIsEnabled(activeGenomeId));
-    }
-
-    const entityId = params.entityId
-      ? buildFocusObjectId({
-          genomeId: genomeIdFromUrl as string,
-          ...parseFocusIdFromUrl(params.entityId)
-        })
-      : null;
-
-    if (entityId && genomeIdFromUrl) {
-      if (entityId !== activeEntityId) {
-        dispatch(
-          updateActiveEntityForGenome({
-            genomeId: genomeIdFromUrl,
-            entityId
-          })
-        );
-      }
-
-      if (activeGenomeId === genomeIdFromUrl && entityId !== activeEntityId) {
-        dispatch(resetFilterPanel());
-      }
+    } else if (genomeId && genomeId !== activeGenomeId) {
+      dispatch(setActiveGenomeId(genomeId)); // FIXME: check this!
 
       entityViewerStorageService.updateGeneralState({
-        activeGenomeId: genomeIdFromUrl
+        activeGenomeId: genomeId
       });
+    }
 
+    if (genomeId && entityId && entityId !== activeEntityId) {
+      dispatch(
+        updateActiveEntityForGenome({
+          genomeId: genomeId,
+          entityId
+        })
+      );
       entityViewerStorageService.updateGeneralState({
-        activeEntityIds: { [genomeIdFromUrl]: entityId }
+        activeEntityIds: { [genomeId]: entityId }
       });
+
+      // if (entityId !== activeEntityId) {
+
+      // }
+
+      // FIXME: where should this belong?
+      // if (activeGenomeId === genomeIdFromUrl && entityId !== activeEntityId) {
+      //   dispatch(resetFilterPanel());
```

Comment:
Can this be done inside `TranscriptsFilter.tsx`?

Reply:
Moved this over to the GeneView component.